### ISSUE TITLE
Plugin not operating on moving window

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#2582]
Several plugins work on the entire simulation domain and not on the moving window only. 
This should be adjusted so that during GPU slides, the output values do not jump.

cc'ing @kossag14

**Comment by ax3l** [comment]
Duplicate of #89.

Generalization of issues #2267, #870.
Also would change what we define as "leaving the box" imho: #517

Depends on #2131.
